esi/lib/networks.py

```diff
@@ -112,3 +112,36 @@ def get_networks_from_port(connection, port, networks_dict={}, floating_ips_dict
         floating_network = connection.network.get_network(floating_network_id)
 
     return parent_network, trunk_networks, trunk_ports, floating_network
+
+
+def create_port(connection, network_name):
+    """
+    Creates a port on the specified network using the network name provided.
+
+    :param connection: An OpenStack connection object
+    :param network_name: The name of the network where the port should be created
+
+    :return: The created port object
+    """
+    network = connection.network.find_network(network_name)
+
+    port = connection.network.create_port(network_id=network.id)
+    return port
+
+
+def attach_floating_ip(connection, floating_ip_address, port_id):
```

Comment:
I think it makes sense to have this function do a little more; otherwise it would probably be easier to have the UI pass in the floating IP and directly call `update_ip` using that. I'd suggest changing the `floating_ip_address` argument into an optional `floating_ip=None` argument. If it's None, create a floating IP.

Reply:
I've structured the UI so that if a user selects to attach a floating IP, the IP is passed, and the port is created and attached to the IP. The port is then passed for provisioning. If no IP is selected, only the network is passed to metalsmith for provisioning, as done previously.That's why I didn't used `None` for floating IP since creating one involves additional check like quota and maybe more.

Reply:
In that case, I don't think you need this specific function; there's simply not enough logic in here to make it needed.